alacritty_terminal/src/keyboard_motion.rs

```diff
@@ -0,0 +1,643 @@
+            {
+                point
+            } else if left {
+                term.semantic_search_left(point)
+            } else {
+                term.semantic_search_right(point)
+            }
+        };
+
+        let mut buffer_point = term.visible_to_buffer(point);
+
+        // Make sure we jump above wide chars
+        buffer_point = expand_wide(term, buffer_point, left);
+
+        // Move to word boundary
+        if !is_boundary(term, buffer_point, left) && left != start {
+            buffer_point = semantic(buffer_point);
+        }
+
+        // Skip whitespace
+        let mut point = advance(term, buffer_point, left);
+        while !is_boundary(term, buffer_point, left) && is_space(term, point) {
+            buffer_point = point;
+            point = advance(term, buffer_point, left);
+        }
+
+        // Assure minimum movement of one cell
+        if !is_boundary(term, buffer_point, left) {
+            buffer_point = advance(term, buffer_point, left);
+        }
+
+        // Move to word boundary
+        if !is_boundary(term, buffer_point, left) && left == start {
+            buffer_point = semantic(buffer_point);
+        }
+
+        // Scroll viewport if necessary
+        scroll(term, buffer_point);
+
+        term.buffer_to_visible(buffer_point).unwrap_or_default().into()
+    }
+
+    /// Move by whitespace separated word, like W/B/E/gE in vi.
+    fn word_move<T: EventListener>(
+        term: &mut Term<T>,
+        point: Point,
+        left: bool,
+        start: bool,
+    ) -> Point {
+        let mut buffer_point = term.visible_to_buffer(point);
+
+        // Make sure we jump above wide chars
+        buffer_point = expand_wide(term, buffer_point, left);
+
+        // Skip whitespace until right before a word
+        if left == start {
+            let mut point = advance(term, buffer_point, left);
+            while !is_boundary(term, buffer_point, left) && is_space(term, point) {
+                buffer_point = point;
+                point = advance(term, point, left);
+            }
+        }
+
+        if left == start {
+            // Skip non-whitespace until right inside word boundary
+            let mut point = advance(term, buffer_point, left);
+            while !is_boundary(term, buffer_point, left) && !is_space(term, point) {
+                buffer_point = point;
+                point = advance(term, buffer_point, left);
+            }
+        } else {
+            // Skip non-whitespace until just beyond word
+            while !is_boundary(term, buffer_point, left) && !is_space(term, buffer_point) {
+                buffer_point = advance(term, buffer_point, left);
+            }
+        };
+
+        // Skip whitespace until right inside word boundary
+        if left != start {
+            while !is_boundary(term, buffer_point, left) && is_space(term, buffer_point) {
+                buffer_point = advance(term, buffer_point, left);
+            }
+        }
```

Comment:
if you look closely on these `if`s you can notice, that  you can safely regroup them to just.

```rust
if left == start {
    // Skip whitespace until right before a word
    let mut point = advance(term, buffer_point, left);
    while !is_boundary(term, buffer_point, left) && is_space(term, point) {
        buffer_point = point;
        point = advance(term, point, left);
    }

    // Skip non-whitespace until right inside word boundary
    let mut point = advance(term, buffer_point, left);
    while !is_boundary(term, buffer_point, left) && !is_space(term, point) {
        buffer_point = point;
        point = advance(term, buffer_point, left);
    }
}

if left != start {
    // Skip non-whitespace until just beyond word
    while !is_boundary(term, buffer_point, left) && !is_space(term, buffer_point) {
        buffer_point = advance(term, buffer_point, left);
    }

    // Skip whitespace until right inside word boundary
    while !is_boundary(term, buffer_point, left) && is_space(term, buffer_point) {
        buffer_point = advance(term, buffer_point, left);
    }
}
```